### PR TITLE
Postprocessing refactor

### PR DESCRIPTION
**Pull request by ffreyer**
Here's a draft for https://github.com/JuliaPlots/Makie.jl/issues/604. 

I moved most of the postprocessing stuff into a new file and wrapped it up in a `PostProcessor` object. Beyond ssao, fxaa and copying I also added an "empty" postprocessor, which doesn't do anything when executed. Depending on some global variables `enable_SSAO` and `enable_FXAA` either the empty or the ssao/fxaa postprocessor ends up in a list in `Screen`. The renderloop then executes postprocessors from that list, in order. 

So if for example `screen.postprocessors[1]` is an empty postprocessor, the renderloop will do nothing instead of the SSAO stuff. This is the case when `enable_SSAO[] = false`. I also added a bit to the `GLVisualizeShader` so that the `position` and `normal` buffers are removed from `fragment_ouput` when `enable_SSAO = false`.

### CODE REVIEW

**Comment by SimonDanisch** [comment]
Nice :) I will need a bit of time to review this ;)

**Comment by ffreyer** [comment]
I might make some more changes this weekend. I noticed, for example, that `render_frame` still attaches all 4 draw buffers even if only two are active/exist.

File: src/gl_backend.jl [hunk 1]
```diff
@@ -45,7 +45,7 @@ function get_texture!(atlas)
 end
 
 # TODO
-const enable_SSAO = Ref(true)
+const enable_SSAO = Ref(false)
```
Comment (SimonDanisch) [inline src/gl_backend.jl:48]: Cool, with all the issue, it seems to be a good idea to only turn it on when actually used!

**Comment by SimonDanisch** [comment]
Let me know once this is ready :)

**Comment by ffreyer** [comment]
I think I'm done with this.

It would be cool to turn on/off ssao (and maybe fxaa) on the fly in the future, i.e. to cause a shader recompilation when the scene demands it. That would also be nice to control `shading` and I think it would be needed for depth peeling. But I didn't want to do this here.

**Comment by ffreyer** [comment]
AbstractPlotting tests pass locally for me, on current `AbstractPlotting#master` (https://github.com/JuliaPlots/AbstractPlotting.jl/commit/95d3375162b047f1ca18f1dfe46a3079ff7747e1)

**Comment by ffreyer** [comment]
`]test GLMakie` fails locally on `idx = N - 1` though it passes here and if I run it manually in a terminal. So I don't think it's an issue. 

Should be ready to merge

**Comment by pbouffard** [comment]
Just gave it a try, works great. Thank you!!

**Comment by SimonDanisch** [comment]
Thanks!

**Pull request merged**